### PR TITLE
Support system-wide dark preference

### PR DESCRIPTION
**Pull request by cassidyjames**
Since it's an OS component, it would make sense to follow this preference

### CODE REVIEW

File: src/PolkitDialog.vala [hunk 1]
```diff
@@ -164,6 +164,17 @@ namespace Ag.Widgets {
...
 
+            const string DESKTOP_SCHEMA = "io.elementary.desktop";
+            const string DARK_KEY = "prefer-dark";
+
+            var lookup = SettingsSchemaSource.get_default ().lookup (DESKTOP_SCHEMA, false);
+
+            if (lookup != null) {
+                var desktop_settings = new Settings (DESKTOP_SCHEMA);
+                var gtk_settings = Gtk.Settings.get_default ();
+                desktop_settings.bind (DARK_KEY, gtk_settings, "gtk_application_prefer_dark_theme", SettingsBindFlags.DEFAULT);
```
Comment (donadigo) [inline src/PolkitDialog.vala]: We prefer having properties have `-` instead of `_` in their names, so `gtk-application-prefer-dark-theme`.

Comment (cassidyjames) [reply src/PolkitDialog.vala]: @donadigo I thought that mapping was done automatically? I'm not creating this property, it's the built-in GTK one.

Comment (donadigo) [reply src/PolkitDialog.vala]: @cassidyjames it's done automatically. What I mean is that GObject uses a naming convention where a property in `Gtk.Settings` of name `gtk_application_prefer_dark_theme` would map to `gtk-application-prefer-dark-theme` when connecting / binding to it.

Comment (cassidyjames) [reply src/PolkitDialog.vala]: @donadigo Ah, I gotcha now. I didn't realize that. I've pushed that plus an update to use a different key after a lot of discussion and thought. :smile:

**Pull request merged**